lucene/core/src/java/org/apache/lucene/analysis/Analyzer.java

```diff
@@ -367,12 +367,12 @@ public void close() {
     /**
      * Original source of the tokens.
      */
-    protected final Consumer<Reader> source;
```

Comment:
Don't change to package private scope here. It will prevent subclasses from outside of the package from accessing those fields (and there may be classes outside of Lucene code doing that).

Reply:
Removing this would break many outside analyzers. I know it's seldom that analyzers access these fields, but this is a real backward breaking change. Don't do this.

I have no problem with the ctors, but here it's serious! Why did Intellij suggest a change like that? Looks like it was not so intelligent. 🤨

Reply:
There's also no risk somebody could do anything wrong. It's both (also next one) final field and the it's for consuming only.

Reply:
I think it's because the field is final and there is a getter for it, so the code analyzer prefers encapsulation?

Reply:
Maybe. It doesn't matter though - this changes the API of a class that's been there for ages. I bet there is a class out there somewhere (let's say A extends Analyzer) and another one (B extends A) where A overrides the getter but B reaches out for the original field. Do we want this to break just to hide a field that can be useful for subclasses just to silence an automatic code inspection? I don't think we should.